src/coreclr/nativeaot/BuildIntegration/Microsoft.NETCore.Native.Unix.targets

```
@@ -47,9 +47,17 @@ The .NET Foundation licenses this file to you under the MIT license.
       <CrossCompileAbi Condition="'$(CrossCompileRid)' == 'linux-arm'">gnueabihf</CrossCompileAbi>
       <CrossCompileAbi Condition="'$(CrossCompileRid)' == 'linux-bionic-arm'">androideabi21</CrossCompileAbi>
       <CrossCompileAbi Condition="'$(CrossCompileRid)' == 'linux-musl-arm'">musleabihf</CrossCompileAbi>
+    </PropertyGroup>
+
+    <!-- alpine's gcc toolchain needs alpine-linux, see https://github.com/llvm/llvm-project/issues/89146 -->
+    <Exec Command="grep -q ID=alpine &quot;$(SysRoot)/etc/os-release&quot;" IgnoreExitCode="true" StandardOutputImportance="Low" Condition="'$(CrossCompileArch)' != ''">
```

Do we require SysRoot to be set anywhere else? Could this break building for Bionic where we don't require SysRoot?

---

We also don't really have grep when running on Windows. This runs on Windows for Bionic.

---

It will just return non-zero exit code in those cases: grep not found, file not found, query returns false etc.

---

It will just return non-zero exit code in those cases: grep not found, file not found, query returned false etc.

---

Would it read `/etc/os-release` from the root if SysRoot is undefined?

---

If SysRoot is undefined and CrossCompileArch is set, then yes. So if someone was on alpine linux trying to publish the app for ubuntu without SysRoot, it will fail. Are there cases where it is possible to cross publish app without specifying SysRoot on Alpine Linux host? In any case, lets make the condition a bit robust

```suggestion
    <Exec Command="grep -q ID=alpine &quot;$(SysRoot)/etc/os-release&quot;" IgnoreExitCode="true" StandardOutputImportance="Low" Condition="'$(CrossCompileArch)' != '' and '$(SysRoot)' != ''">
```